blivet/zfcp.py

```diff
@@ -47,62 +49,218 @@ def logged_write_line_to_file(fn, value):
 zfcpconf = "/etc/zfcp.conf"
 
 
-class ZFCPDevice:
+def _is_lun_scan_allowed():
+    """Return True if automatic LUN scanning is enabled by the kernel."""
+
+    allow_lun_scan = util.get_kernel_module_parameter("zfcp", "allow_lun_scan")
+    return allow_lun_scan == "Y"
+
+
+def _is_port_in_npiv_mode(device_id):
+    """Return True if the device ID is configured in NPIV mode. See
+    https://www.ibm.com/docs/en/linux-on-systems?topic=devices-use-npiv
     """
-        .. warning::
-            Since this is a singleton class, calling deepcopy() on the instance
-            just returns ``self`` with no copy being created.
+
+    port_in_npiv_mode = False
+    port_type_path = "/sys/bus/ccw/devices/{}/host*/fc_host/host*/port_type".format(device_id)
+    port_type_paths = glob.glob(port_type_path)
+    try:
+        for filename in port_type_paths:
+            with open(filename) as f:
+                port_type = f.read()
+            if re.search(r"(^|\s)NPIV(\s|$)", port_type):
+                port_in_npiv_mode = True
+    except OSError as e:
+        log.warning("Couldn't read the port_type attribute of the %s device: %s", device_id, str(e))
+        port_in_npiv_mode = False
+
+    return port_in_npiv_mode
+
+
+def is_npiv_enabled(device_id):
+    """Return True if the given zFCP device ID is configured and usable in
+    NPIV (N_Port ID Virtualization) mode.
```

Comment:
"""Return True if the given zFCP device ID is configured in
    NPIV (N_Port ID Virtualization) mode and zfcp auto LUN scan is not disabled.